Justfile

```diff
@@ -0,0 +1,91 @@
+# when false, disables code coverage
+coverage := "true"
+# the output directory for the documentation
+docsDir  := "docs/html"
+# the Antora playbook file to use when building docs
+playbook := "docs/antora-playbook-local.yml"
+# variables for running containerized jenkins
+container := "jenkins" # the name of the container
+port      := "8080"    # the port to forward Jenkins to
+
+# describes available recipes
+help: 
+  just --list --unsorted
+
+# wipe local caches
+clean: 
+  ./gradlew clean 
+  rm -rf {{docsDir}}
+
+# Run unit tests
+test class="*":
+  #!/usr/bin/env bash
+  set -euxo pipefail
+  coverage=$([[ {{coverage}} == "true" ]] && echo "jacocoTestReport" || echo "")
+  ./gradlew test --tests '{{class}}' $coverage
+
+# Run spotless & codenarc
+lint: 
+  ./gradlew spotlessApply codenarc
+
+# Build the JPI
+jpi: 
+  ./gradlew clean jpi 
+
+# executes the CI checks (test lint jpi)
+ci: test lint jpi
+
+# Build the local Antora documentation
+docs: 
```

Comment:
tried running this locally:
Unable to find image 'docker.pkg.github.com/boozallen/sdp-docs/builder:latest' locally
docker: Error response from daemon: Head https://docker.pkg.github.com/v2/boozallen/sdp-docs/builder/manifests/latest: no basic auth credentials.
I have ~/.git-credentials for github.

Reply:
`docker login docker.pkg.github.com` and then use your GH user/pat

we could try to automate that part too

Reply:
hmm did not work
```
carlosokieffe@C02D80XNMD6V templating-engine-plugin % docker login docker.pkg.github.com                              
Username: cokieffebah
Password: 

Login Succeeded
carlosokieffe@C02D80XNMD6V templating-engine-plugin % 
carlosokieffe@C02D80XNMD6V templating-engine-plugin % git pull docker.pkg.github.com/boozallen/sdp-docs/builder:latest
fatal: 'docker.pkg.github.com/boozallen/sdp-docs/builder:latest' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
carlosokieffe@C02D80XNMD6V templating-engine-plugin % just docs                                                       
docker run -it --rm -v ~/.git-credentials:/home/antora/.git-credentials -v $(pwd):/app -w /app docker.pkg.github.com/boozallen/sdp-docs/builder generate --generator booz-allen-site-generator --to-dir docs/html docs/antora-playbook-local.yml
Unable to find image 'docker.pkg.github.com/boozallen/sdp-docs/builder:latest' locally
docker: Error response from daemon: unauthorized: Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.
See 'docker run --help'.
error: Recipe `docs` failed on line 47 with exit code 125
```

Reply:
I approved the PR.  just noting the issues with the docs.

Reply:
you accidentally ran `git pull` instead of `docker pull`. 

when you authenticated, did you use your password or a PAT? 

if a PAT, it would need to have the `read:packages` scope